src/engine/GroupBy.cpp

```diff
@@ -1013,6 +1014,7 @@ GroupBy::isSupportedAggregate(sparqlExpression::SparqlExpression* expr) {
   if (auto val = dynamic_cast<GroupConcatExpression*>(expr)) {
     return H{GROUP_CONCAT, val->getSeparator()};
   }
+  if (dynamic_cast<StdevExpression*>(expr)) return H{STDEV};
```

Comment:
This is wrong, 
There is now way that you can lazily compute a StdDev or using a HashMap, as you always need the full input the first compute the average, and then combine the average with all the other inputs.
So you just can remove those lines and let this yield `nullopt` as it was (maybe add a comment `NOTE: The STDEV function is not suitable for lazy and hash map optimizations.`